api/desecapi/pdns_change_tracker.py

```diff
@@ -225,11 +236,14 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 change.api_do()
                 if change.axfr_required:
                     axfr_required.add(change.domain_name)
-            except Exception as e:
-                # TODO gather as much info as possible
-                #  see if pdns and api are possibly in an inconsistent state
+            except PDNSValidationError as e:
                 self.transaction.__exit__(type(e), e, e.__traceback__)
                 raise e
+            except Exception as e:
+                self.transaction.__exit__(type(e), e, e.__traceback__)
+                exc = ValueError('For changes %s, %s occured when applying %s' %
```

Comment:
Should be `RuntimeError`

Reply:
Multiple tests fail when this is `RuntimeError` (see commit history)